be/src/storage/lake/update_compaction_state.cpp

```diff
@@ -74,7 +78,7 @@ Status CompactionState::_load_segments(Rowset* rowset, const TabletSchemaCSPtr&
     auto& dest = pk_cols[segment_id];
     auto col = pk_column->clone();
     if (itr != nullptr) {
-        const auto num_rows = rowset->num_rows();
+        const auto num_rows = rowset->num_rows() / rowset->num_segments();
```

Comment:
Can we remove the vector reserve here if we don't know the exact number of rows? Is there any rationale or benchmark demonstrating that doing a potentially inaccurate reserve improves performance or reduces memory usage?

Reply:
I haven't do any test here, but I think i can remove `reserve` here.

Reply:
Done